Jenkinsfile

```diff
@@ -91,6 +96,21 @@ pipeline {
 						ant -f eclipse.platform.swt/bundles/org.eclipse.swt/buildSWT.xml new_build_with_create_file -DTAG=HEAD
 					'''
 				}
+				script { // Stash jdk-headers for transfer to the natives build machines
+					if ( params.forceNativeBuilds || (fileExists ('tmp/build_changed.txt') && fileExists ('tmp/natives_changed.txt')) ){
+						dir('jdkHeaderDownloads') {
+							for(os in [ 'macosx', 'linux', 'win32' ]) {
+								sh """
+									curl ${getJDKHeaderProviderURL(os)} -o jdk.tar.gz
+									tar -xzf jdk.tar.gz ${ os == 'win32' ? './include/': 'include/'}
```

Comment:
@merks can you tell why in the Windows JDKs available at `https://download.eclipse.org/justj/jres/17/downloads/20230204_0657/` there is always an extra folder named `.` in the tar.gz root which contains the jdk content?
For Linux and Macos the jdk is always contained directly in the zip's root.

Reply:
I think it's produced the same way on all platforms by this:

https://github.com/eclipse-justj/justj/blob/400371be17b1e5b4819ef1d977cb4133d7bbf277/releng/org.eclipse.justj.releng/build-jre.sh#L373-L376

It should be logged here:

https://ci.eclipse.org/justj/job/build-jres/248/

But there I see the following as if the * is expanded to "."  which seems rather strange behavior.
```
+ tar -czf ../org.eclipse.justj.openjdk.hotspot.jre.full-17.0.6-win32-x86_64.tar.gz .
```
It should be possible to run this script locally and reproduce such a problem.   But I won't have time today or tomorrow to investigate that.

Reply:
Found time for a quick look...  

It's this that is repackaging after signing all the executables and dlls:

https://github.com/eclipse-justj/justj/blob/400371be17b1e5b4819ef1d977cb4133d7bbf277/releng/org.eclipse.justj.releng/Jenkinsfile#L513

I thing that should be '*' and not '.' to avoid this problem.

Reply:
I've kicked off a build to build new JREs that should fix this problem...

BTW, this file contains relative URIs of all the latest Java 17 JREs:

https://download.eclipse.org/justj/jres/17/downloads/latest/justj.manifest

So in principle one could use that to avoid hard coding the specific URL of a current latest...

Reply:
> I've kicked off a build to build new JREs that should fix this problem...

Thanks a lot! The windows tar.gz at https://download.eclipse.org/justj/www/download.eclipse.org.php?file=jres/17/downloads/20230423_0633 look perfectly fine now!



> I've kicked off a build to build new JREs that should fix this problem...
> 
> BTW, this file contains relative URIs of all the latest Java 17 JREs:
> 
> https://download.eclipse.org/justj/jres/17/downloads/latest/justj.manifest
> 
> So in principle one could use that to avoid hard coding the specific URL of a current latest...

Thanks for that hint. I was already wondering if there is a latest URL from which one can just download them directly.
But since actually only the C header files provided by a JDK (in the `include`) are of interest in this case and I believe that they don't change often I was fine with how it is.
But I'll have a look so that always the latest tar.gz are used.